config/http_config.go

```diff
@@ -878,8 +978,32 @@ func cloneRequest(r *http.Request) *http.Request {
 	return r2
 }
 
+type tlsConfigOptions struct {
+	secretManager SecretManager
+}
+
+// TLSConfigOption defines an option that can be applied to the HTTP client.
+type TLSConfigOption func(options *tlsConfigOptions)
+
+// WithSecretManager allows setting the secret manager.
+func withSecretManager(manager SecretManager) TLSConfigOption {
```

Comment:
Shouldn't this be:

```suggestion
func WithSecretManagerTLSConfigOption(manager SecretManager) TLSConfigOption {
```
(for dedup and to be exported)?

Reply:
We can do this with a common object that implements both interfaces. What do you think of the new solution?

Reply:
Whatever is simpler here